Sign up screen and Login screen forget screen data passing

TajammalCodeLab opened the pull request.

Commit:
customised the Login screen

I have customised the Login screen to match the UI

Commit:
Added colors, icons, worked on signUP

added custom colors, Icons, worked on signup screen, added scroll view inside the signup container

Commit:
added signUp page, Custom class, base controller, protectd and upproted logic

Commit:
Update UserInterfaceState.xcuserstate

Commit:
Update UserInterfaceState.xcuserstate

Commit:
Update UserInterfaceState.xcuserstate

Commit:
woked on overall code

Commit:
Update UserInterfaceState.xcuserstate

Commit:
add animation,

Commit:
Redo

Commit:
Update UserInterfaceState.xcuserstate

Commit:
implemented validation on Email and password

Commit:
Add foget password screen, dashboard, sending data through protocal, UI complete

mianusamaswl commented on signInAndSignOut_feature/UI/DashboardController/DashboardViewController.swift:
```diff
@@ -0,0 +1,47 @@
+
+import UIKit
+
+class DashboardViewController: UIViewController, DataSendingProtocol {
```
Instead of Inheriting the DashboardViewController from DataSendingProtocol 
Use extension out side the body of class to inheriate the Dashbaord from DataSendingProtocl

mianusamaswl commented on signInAndSignOut_feature/UI/SignUpController/SignUpViewController.swift:
```diff
@@ -7,26 +7,101 @@
+    }
+    
+    private func directToSingInScreen(){
+        let vc = self.main.instantiateViewController(identifier: Identifiers.LOGIN_ID)
```
Use Pop back to get back to the login screen Instead of using this

mianusamaswl commented on signInAndSignOut_feature/UI/LoginController/LoginViewController.swift:
```diff
@@ -6,133 +6,166 @@
-        layoutUI()
-        //addBorder() // Adding the border
+
+protocol DataSendingProtocol {
```
Create a separate file and define protocols on that file.

Commit:
PR "Removed access comments ", "Mad few changes in code"

Removed access comments throughout gout the file only keep the necessary comments to understand the flow
deleted this file "CustomTextFieldViewController.swift"
Used extension inherited riate the Dashbaord from DataSendingProtocol
Use Pop back to get back to the login screen Instead push
Created a separate file and defined protocols on that file.

TajammalCodeLab commented:
Removed access comments throughout gout the file only keep the necessary comments to understand the flow
deleted this file "CustomTextFieldViewController.swift"
Used extension inherited riate the Dashbaord from DataSendingProtocol
Use Pop back to get back to the login screen Instead push
Created a separate file and defined protocols on that file.

Commit:
Adde session handling feature implemented using userDefualt
